internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -515,6 +545,8 @@ func rollbackInstall(ctx context.Context, log *logger.Logger, topDirPath, versio
 	return nil
 }
 
+var copyActionStoreFunc = copyActionStore // abstraction for testability
```

Comment:
same as with previous occurrence
i don't know if this is just a matter of my personal taste, but i don't like this pattern